presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -170,12 +172,28 @@ private Expression simplifyExpression(Expression expression)
             }
             expression = ExpressionTreeRewriter.rewriteWith(new PushDownNegationsExpressionRewriter(), expression);
             expression = ExpressionTreeRewriter.rewriteWith(new ExtractCommonPredicatesExpressionRewriter(), expression, NodeContext.ROOT_NODE);
+            expression = ExpressionTreeRewriter.rewriteWith(new ReplaceSymbolEqualsSymbolWithIsNotNullPredicate(), expression);
             Map<NodeRef<Expression>, Type> expressionTypes = getExpressionTypes(session, metadata, sqlParser, types, expression, emptyList() /* parameters already replaced */);
             ExpressionInterpreter interpreter = ExpressionInterpreter.expressionOptimizer(expression, metadata, session, expressionTypes);
             return LiteralInterpreter.toExpression(interpreter.optimize(NoOpSymbolResolver.INSTANCE), expressionTypes.get(NodeRef.of(expression)));
         }
     }
 
+    private static class ReplaceSymbolEqualsSymbolWithIsNotNullPredicate
+            extends ExpressionRewriter<Void>
+    {
+        @Override
+        public Expression rewriteComparisonExpression(ComparisonExpression node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+        {
+            if (node.getType() == EQUAL && node.getLeft() instanceof SymbolReference && node.getRight() instanceof SymbolReference
```

Comment:
maybe instead of checking just only symbol reference you could check the entire expression, to cover things like:
`(a + b) = (a + b)` then it would become `(a + b) IS NOT NULL` and it could become `a IS NOT NULL AND b IS NOT NULL`